components/image-builder-mk3/pkg/orchestrator/orchestrator.go

```diff
@@ -624,6 +624,9 @@ func (o *Orchestrator) getAbsoluteImageRef(ctx context.Context, ref string, allo
 		}
 		return "", status.Error(codes.Unauthenticated, "cannot resolve image")
 	}
+	if resolve.TooManyRequestsMatcher(err) {
+		return "", status.Errorf(codes.Unavailable, "upstream registry responds with 'too many request': %v", err)
+	}
 	if err != nil {
 		return "", status.Errorf(codes.Internal, "cannot resolve image: %v", err)
```

Comment:
Code LGTM 🚀 

For further reviewers, this is how we detect `imageBuildFailedUser` https://github.com/gitpod-io/gitpod/blob/4524c19868e69c920c0c44e65b5dea3268143e5b/components/server/src/workspace/workspace-starter.ts#L1300-L1306

and error message was
```
13 INTERNAL: cannot resolve image: httpReadSeeker: failed open: unexpected status code https://registry-1.docker.io/v2/gitpod/workspace-full/xxx: 429 Too Many Requests - Server message: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
```